lib/base.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+var path = require('path');
+var invalid = require('./invalid');
+var utils = require('./utils');
+var once = require('./once');
+
+/**
+ * Copy a file from `src` to `dest`
+ *
+ * @param {String} `src` Source filepath
+ * @param {String} `dest` Destination filepath
+ * @param {Object} `options`
+ * @param {Function} `cb` Callback function
+ * @api public
+ */
+
+function copyBase(src, dest, options, callback) {
+  if (typeof options !== 'object') {
+    callback = options;
+    options = {};
+  }
+
+  if (arguments.length < 3) {
+    return invalid.apply(null, arguments);
+  }
+
+  var fs = utils.fs; // graceful-fs (lazyily required)
+  var opts = utils.extend({overwrite: true}, options);
+  var cb = once(callback);
+  var listener = once(ws);
+  src = path.resolve(src);
+
+  var rs = fs.createReadStream(src)
+    .on('error', handleError('read', src, opts, cb))
+    .on('readable', listener)
+    .on('end', listener);
```

Comment:
`.on('end')` might not be needed (I'm not sure how to test this) but it probably doesn't matter since `listener` is wrapped with `once`